scipy/spatial/transform/rotation.py

```diff
@@ -1590,3 +1590,202 @@ def __call__(self, times):
+
+
+class Spline(object):
+    """Interpolate rotations using cubic splines.
+
+    The interpolation between consecutive rotations is done using cubic splines
+    [1]_ to ensure that the angular velocity and accleration at the end of a
+    current interval are equal to the velocity and accleration at the start of
+    the next interval. This leads to smooth overall change in orientation.
+
+    Parameters
+    ----------
+    times : array_like, shape (N,)
+        Times of the known rotations. At least 2 times must be specified.
+    rotations : `Rotation` instance
+        Rotations to perform the interpolation between. Must contain N
+        rotations.
+    omega_i : array_like, shape (3,), optional
+        Initial angular velocity in radians at the time of the first
+        orientation. Default is `[0, 0, 0]`.
+    omega_f : array_like, shape (3,), optional
+        Final angular velocity in radians at the time of the last orientation.
+        Default is `[0, 0, 0]`.
+
+    References
+    ----------
+    .. [1] `Quaternion Spline
+            <http://qspline.sourceforge.net/qspline.pdf>`_
+    """
+    def __init__(self, times, rotations, omega_i=[0, 0, 0], omega_f=[0, 0, 0]):
+        if len(rotations) == 1:
+            raise ValueError("`rotations` must contain at least 2 rotations.")
+
+        times = np.asarray(times)
+        if times.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(times.ndim))
+
+        if times.shape[0] != len(rotations):
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of timestamps given, got {} rotations "
+                             "and {} timestamps.".format(
+                                len(rotations), times.shape[0]))
+
+        omega_i = np.asarray(omega_i)
+        if omega_i.shape != (3,):
+            raise ValueError("Expected initial angular velocity `omega_i` to "
+                             "be specified as a 1 dimensional 3-vector, got "
+                             "shape {}.".format(omega_i.shape))
+
+        omega_f = np.asarray(omega_f)
+        if omega_f.shape != (3,):
+            raise ValueError("Expected final angular velocity `omega_f` to "
+                             "be specified as a 1 dimensional 3-vector, got "
+                             "shape {}.".format(omega_f.shape))
+        self.t = times
+        self.dt = np.diff(times)
+        if np.any(self.dt <= 0):
+            raise ValueError("Times must be in strictly increasing order.")
+
+        # Calculate intermediate angular velocities
+        dt_inv = 1 / self.dt
+        rotvecs = (rotations[:-1].inv() * rotations[1:]).as_rotvec()
+        w_int, diff = _intermediate_step(np.zeros(len(rotations) - 2, 3),
+                                         rotvecs, dt_inv)
+
+        while diff >= 1e-6:
```

Comment:
Should this 1e-6 magic number be configurable? and why is 1e-6 the right number?

Reply:
This `1e-6` was mentioned in the paper which Nikolay provided. I'm not sure it suits all applications. Should I add a tolerance argument to the function and explain the convergence steps in the docstring?